client/src/app/containers/Game/PegAction.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import Button from '@material-ui/core/Button';
+import ButtonGroup from '@material-ui/core/ButtonGroup';
+import SendIcon from '@material-ui/icons/Send';
+
+import { ActionInputProps } from './types';
+import { useGame } from './useGame';
+
+const PegAction: React.FunctionComponent<ActionInputProps> = ({
+    isBlocking,
+}) => {
+    const { clearSelectedCards, selectedCards, submitPegAction } = useGame();
+
+    const handleClick = async () => {
+        await submitPegAction({ selectedCards });
+        clearSelectedCards();
+    };
+
+    return (
+        <ButtonGroup
+            orientation='vertical'
+            color='primary'
+            aria-label='vertical outlined primary button group'
+        >
+            <Button
+                disabled={!isBlocking}
+                color='secondary'
+                // TODO it's probably semantically better if we also have a submitSayGoAction
+                onClick={() => submitPegAction({ selectedCards: [] })}
```

Comment:
:mag: yes.

Reply:
Again, see my comment above. This is a shortcoming of how I've defined `createActionHandler`. This one's a bit different - the reason it's not "easy" to do this is because the server has no notion of a say go action; just a peg action that looks slightly different. Again, need to invest some time into thinking of a better way.